components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -163,6 +193,17 @@ func (r *NotebookReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		}
 	}
 
+	events, err := r.eventLister.Events(req.Namespace).List(labels.Everything())
```

Comment:
How should we avoid forwarding events that belongs to other notebooks or CRs?

Reply:
Good catch, I totally disregarded both cases. Events are named like `<nbName>-xxxxxxx`, statefulsets are named like `<nbName>` and pods (created by the previous sts) like `<nbName>-[0-9]+`. The `req.Name` in `Reconcile` is equal to the name of the observed event (pod, generic event). What if we filter the events that have an `event.InvolvedObject.Name` starting with the sts name? Or is this solution making a lot of assumptions about naming conventions?

Another solution I considered is retrieving the event via `r.Get(...)` (to avoid listing and filtering), the problem I had is that, some events occur before the notebook `instance` had been instantiated, so I could no reissue those events as that `instance` events...

Reply:
This is solved in https://github.com/kubeflow/kubeflow/pull/4139/commits/d032910ce8cee317e09193a108fea4c785110aa1. Now, if the reconcile request was originated by a watched event and this event is related to a nb pod or sts, the event is reissued. This also avoids sending the same event multiple times.